Ginger/GingerCoreNETUnitTest/SolutionTestsLib/VaribaleRepositorySerializerTest.cs

```diff
@@ -0,0 +1,154 @@
+#region License
+/*
+Copyright © 2014-2019 European Support Limited
+
+Licensed under the Apache License, Version 2.0 (the "License")
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at 
+
+http://www.apache.org/licenses/LICENSE-2.0 
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS, 
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+See the License for the specific language governing permissions and 
+limitations under the License. 
+*/
+#endregion
+
+using amdocs.ginger.GingerCoreNET;
+using Amdocs.Ginger.Common;
+using Amdocs.Ginger.CoreNET.Repository;
+using Amdocs.Ginger.Repository;
+using GingerCore;
+using GingerCore.Variables;
+using GingerCoreNETUnitTest.RunTestslib;
+using GingerTestHelper;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace GingerCoreNETUnitTests.SolutionTestsLib
+{
+   
+    [Level1]
+    [TestClass]
+    public class VaribaleRepositorySerializerTest
+    {
+        public static NewRepositorySerializer RS = new NewRepositorySerializer();
+        public static BusinessFlow BusinessFlow = new BusinessFlow();
+        static SolutionRepository SR;
+        static ObservableList<VariableBase> varList = new ObservableList<VariableBase>();
+        VariableBase var = null;
+
+        [ClassInitialize]
+        public static void ClassInitialize(TestContext TC)
+        {
+            string TempSolutionFolder = TestResources.GetTestTempFolder(@"Solutions"+ Path.DirectorySeparatorChar+ "Variables");
+            if (Directory.Exists(TempSolutionFolder))
+            {
+                Directory.Delete(TempSolutionFolder, true);
+            }
+            SR = GingerSolutionRepository.CreateGingerSolutionRepository();
+
+            SR.CreateRepository(TempSolutionFolder);
+            NewRepositorySerializer.AddClassesFromAssembly(typeof(BusinessFlow).Assembly);
+            SR.Open(TempSolutionFolder);
+            string FileName = TestResources.GetTestResourcesFile(@"Solutions"+ Path.DirectorySeparatorChar+ "Variables" + Path.DirectorySeparatorChar + "BusinessFlows" + Path.DirectorySeparatorChar + "Business Flow 1.Ginger.BusinessFlow.xml");
+
+            BusinessFlow = (BusinessFlow)RS.DeserializeFromFile(FileName);
+            varList = BusinessFlow.GetVariables();
+        }
+
+        [TestMethod]
+        public void VariablesCount()
+        {
+            // Assert   
+            Assert.AreEqual(8, BusinessFlow.Variables.Count, "Variable Count");
+        }
+
+        [TestMethod]
+        public void VariableString()
+        {
+            foreach (VariableBase variable in varList)
```

Comment:
please split it to different unit tests. Because let's say even single variable load will fail it will fail the whole unit test.
Ideally 1 unit should test 1 functionality